src/rpc/blockchain.cpp

```diff
@@ -811,6 +812,180 @@ UniValue getblock(const UniValue& params, bool fHelp)
+        );
+
+    LOCK(cs_main);
+
+    std::string strHash = params[0].get_str();
+
+    if (!fMaturityHeightIndex)
+    {
+        throw JSONRPCError(RPC_TYPE_ERROR, "maturityHeightIndex option not set: can not retrieve info");
+    }
+
+    // If height is supplied, find the hash
+    if (strHash.size() < (2 * sizeof(uint256))) {
+        // std::stoi allows characters, whereas we want to be strict
+        regex r("[[:digit:]]+");
+        if (!regex_match(strHash, r)) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block height parameter");
+        }
+
+        int nHeight = -1;
+        try {
+            nHeight = std::stoi(strHash);
+        }
+        catch (const std::exception &e) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block height parameter");
+        }
+
+        if (nHeight < 0 || nHeight > chainActive.Height()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Block height out of range");
+        }
+        strHash = chainActive[nHeight]->GetBlockHash().GetHex();
+    }
+
+    uint256 hash(uint256S(strHash));
+
+    int verbosity = 1;
+    if (params.size() > 1) {
+        if(params[1].isNum()) {
+            verbosity = params[1].get_int();
+        } else {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Verbosity must be in range from 1 to 2");
+        }
+    }
+
+    if (verbosity < 1 || verbosity > 2) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Verbosity must be in range from 1 to 2");
+    }
+
+    if (mapBlockIndex.count(hash) == 0)
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
+
+    CBlock block;
+    CBlockIndex* pblockindex = mapBlockIndex[hash];
+
+    if (fHavePruned && !(pblockindex->nStatus & BLOCK_HAVE_DATA) && pblockindex->nTx > 0)
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Block not available (pruned data)");
+
+    if(!ReadBlockFromDisk(block, pblockindex))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Can't read block from disk");
+
+    UniValue blockJSON = blockToJSON(block, pblockindex, verbosity >= 2);
+    
+    //Add certificates that became mature with this block    
+    if (block.nVersion == BLOCK_VERSION_SC_SUPPORT)
+    {
+        UniValue matureCertificate(UniValue::VARR);
+
+        int height = pblockindex->nHeight;
+        if (pblocktree == NULL)
+        {
+            throw JSONRPCError(RPC_TYPE_ERROR, "DB not initialized: can not retrieve info");
+        }
+        std::vector<CMaturityHeightKey> matureCertificatesKeys;
+        pblocktree->ReadMaturityHeightIndex(height, matureCertificatesKeys);
+        for (const CMaturityHeightKey& key : matureCertificatesKeys) 
+        {
+            if (verbosity == 2)
+            {
+                UniValue objCert(UniValue::VOBJ);
+                CScCertificate certAttempt;
+                uint256 hashBlock{};    
+                if (GetCertificate(key.certId, certAttempt, hashBlock, true))
```

Comment:
Instead of allowSlow=true wouldn't be better to require enabling the txIndex when maturityheightindex is on?